app/__tests__/api/chat/route.test.ts

```diff
@@ -1,20 +1,24 @@
-import { testApiHandler } from 'next-test-api-route-handler';
-import { POST } from './route'; 
+import { POST } from '../../../api/chat/route';
 
 describe('POST /api/chat', () => {
   it('checks if the API works', async () => {
-    await testApiHandler({
-      handler: POST,
-      test: async ({ fetch }) => {
-        const res = await fetch({
-          method: 'POST',
-          headers: { 'Content-Type': 'application/json' },
-          body: JSON.stringify({
-            messages: [{ role: 'user', content: 'Hello' }],
-          }),
-        });
-        expect(res.status).toBe(200); 
-      },
+    // Create a mock Request object
```

Comment:
Jika masih ingin menguji dari perspektif pengguna (end-to-end), bisa mempertahankan pengujian dengan `fetch()` di tambahan test case lain.
```suggestion
    global.fetch = jest.fn(() =>
        Promise.resolve({
            json: () => Promise.resolve({
                aiResponse: 'Mocked response',
                metadata: { finishReason: 'stop', usage: { promptTokens: 10, completionTokens: 20 } }
            }),
            status: 200,
        })
    ) as jest.Mock;
```